src/Xamarin.Android.Build.Tasks/Tasks/Desugar.cs

```diff
@@ -85,6 +85,8 @@ protected override string GenerateCommandLineCommands ()
 			foreach (var jar in InputJars) {
 				var output = Path.Combine (OutputDirectory, BitConverter.ToString (md5.ComputeHash (Encoding.UTF8.GetBytes (jar))) + Path.GetFileName (jar));
 				outputs.Add (output);
+				cmd.AppendSwitch ("--classpath_entry ");
+				cmd.AppendFileNameIfNotNull (jar);
```

Comment:
Doesn't this need to be a `class-path` directory? This is just passing the current jar file. What will this do if the dependencies are in another directory?

Reply:
@jonathanpeppers ^ ?

Reply:
never mind. Just noticed we do this all in one hit. So there will be multiple `--classpath_entry` lines on the command line. So it should be able to find all the required dependencies. 

That does however raise the question of hitting the command line length limit on Windows...

Reply:
Interesting .. I wonder if 

`java -jar <pathto>/desugar_deploy.jar @foo.txt`

will work. where `foo.txt` is the list of arguments.

Reply:
@erikpowa FYI I'm going to add some commits to this.

I'll check the response file thing, and add a test to this using the repro from: https://github.com/xamarin/xamarin-android/issues/2143 

I'd rather us have a test with the offending jar file showing this is broken, so we'll know it still works when R8/D8 are added.

Reply:
@jonathanpeppers sounds great.